server/routes/updateBanner.ts

```diff
@@ -0,0 +1,49 @@
+import { Response, NextFunction } from 'express';
+import { TypedRequestBody } from '../types';
+import { DB } from '../database';
+import validateChain from '../util/validateChain';
+import validateRoles from '../util/validateRoles';
+import { CommunityBannerInstance } from '../models/community_banner';
+
+enum UpdateBannerErrors {
+  NoChain = 'Must supply a chain ID',
+  NoPermission = `You do not have permission to update banner`,
+}
+
+
+type UpdateBannerReq = Omit<CommunityBannerInstance, 'id'> & {
+  chain_id: string;
+  banner_text: string;
+};
+
+const updateBanner = async (
+  models: DB,
+  req: TypedRequestBody<UpdateBannerReq>,
+  res: Response,
+  next: NextFunction
+) => {
+  const [chain, error] = await validateChain(models, req.body);
+  if (error) return next(new Error(UpdateBannerErrors.NoChain));
+  const isAdmin = validateRoles(models, req.user, 'admin', chain.id);
+  if (!isAdmin) return next(new Error(UpdateBannerErrors.NoPermission));
+
+  const { banner_text } = req.body;
```

Comment:
We should verify it isn't null/undefined, unless we want to make it deletable (is there any way to delete a banner?).

Reply:
yeah, i didn't write a deleteBanner route for v1, but i'm going to before deploying this. however, i'm not concerned if this is null bc the ui can also catch this and not display anything.

Reply:
What happens if a user makes a raw request with a null value? The idea is to have defined behavior for all possible calls regardless of UI.

Reply:
put in a solution for this, but I'm not sure you'll like it 😛